File: Baekjoon/5073 - 삼각형과 세 변/5073.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int triangle[3];
    while(1) {
        for(int i = 0; i < 3; i++) cin >> triangle[i];
        if(!triangle[0]) break;
        sort(triangle, triangle+3);
        if(triangle[0]+triangle[1] <= triangle[2]) cout << "Invalid\n";
        else if(triangle[0] == triangle[2]) cout << "Equilateral\n";
        else if(triangle[0] == triangle[1]) cout << "Isosceles\n";
        else if(triangle[1] == triangle[2]) cout << "Isosceles\n";
        else cout << "Scalene\n";
    }
}

File: solution/05xxx/5073 - 삼각형과 세 변/5073.py
import sys
input = sys.stdin.readline

while True:
    triangle = sorted(list(map(int, input().split())))

    if triangle[2] == 0:
        break
    elif triangle[0] + triangle[1] <= triangle[2]:
        print("Invalid")
    elif triangle[0] == triangle[2]:
        print("Equilateral")
    elif triangle[0] == triangle[1] or triangle[1] == triangle[2]:
        print("Isosceles")
    else:
        print("Scalene")